tasks/main.yml

```diff
@@ -156,8 +170,8 @@
         disk_format: "{{ item.type | default('qcow2') }}"
         filename: "{{ os_images_cache }}/{{ item.0.name }}/{{ item.0.name }}.{{ item.type | default('qcow2') }}"
         properties: "{{ os_images_common_properties | combine(item.0.properties | default({})) or omit }}"
-        kernel: "{{ item.1.id if 'baremetal' in item.0.elements else omit }}"
-        ramdisk: "{{ item.2.id if 'baremetal' in item.0.elements else omit }}"
+        kernel: "{{ item.1.id if item.0.elements is defined and 'baremetal' in item.0.elements else omit }}"
+        ramdisk: "{{ item.2.id if item.0.elements is defined and 'baremetal' in item.0.elements else omit }}"
```

Comment:
Protip: When variable expressions get busy like this, you can use task-local variables to clean things up.

```
  ramdisk: "{{ item.2.id if is_baremetal else omit }}"
vars:
  is_baremetal: "{{ item.0.elements is defined and 'baremetal' in item.0.elements }}"
```